### ISSUE TITLE
[Feature Request] Add more graphs

### ISSUE THREAD

**Issue by bloedboemmel** [Yooooomi/your_spotify#358]
First all: Amazing project, especially now with the performance boost :)

I think it only needs a little more graphs. Spotify can give you a lot of audio features for every song and i would be great to display them in a simple graph (maybe in the same look like "Artist listening distribution" or "Best artists for hour of day")
You can see that in [their own example](https://developer.spotify.com/documentation/web-api/reference/get-audio-features):
![image](https://github.com/Yooooomi/your_spotify/assets/58067119/44fe79da-eb9c-4e55-9840-5ccf2c0bae0c)
I am not sure, if you are already saving the audio-features in the db, but i think it wouldn't affect the performance a lot.

Also I thought about adding a world map, which is showing your current "listening map" 
In my head it looks a little bit like that (but of course it's adjusted to the design of your_spotify):
![image](https://github.com/Yooooomi/your_spotify/assets/58067119/3bdb50d5-e6f9-4738-894d-41cc8e4c6aca)

Let me know what you think about this feature and I will try to implement it

**Comment by Yooooomi** [comment]
Hello, I think it's a good idea!  The feature would not be available before having synced the whole track database, which for me represents more than 100k tracks played.
After that, stats are pretty simple to display. However I think we would need another way to display stats, as just piling up stats in the "All stats" page will not age well.

**Comment by bloedboemmel** [comment]
I totally agree with the arrangement of the "All stats". It would get too overloaded and also would take a long time to load.

**Comment by bloedboemmel** [comment]
After a quick research I found out that you can't get the country or location of the tracks or artists. So it wouldn't be possible to create the map (at least when using only the Spotify API)

**Comment by Yooooomi** [comment]
I'm still thinking about how could I change the UX a bit to display more stats or in a maybe less robotic way. If you ever cross cool websites or already have ideas, feel free to post them here  :)

**Comment by bloedboemmel** [comment]
I already added the migrations and api-calls for the feature in my fork.
I also tired to add the audio-features to the trackStats page, but right it only shows the numbers.

Maybe we can show energy, danceability and valence in bars from 0 to 1 (or valence as a emoji, but i think that wouldn't go well with the overall design)

BPM should be shown as a number and the numerical key should be shown in the corresponding pitch.

For the graphs in the overall stats we can determine, if the songs are more likely live (liveness),
If they are more intrumentalness or have more speechiness
We can map the average tempo per hour. Or the average mood per hour.

These are just some ideas, but I also couldn't find any graphs

**Comment by bloedboemmel** [comment]
I also found some interseting graphs [here](https://towardsdatascience.com/spotify-api-audio-features-5d8bcbd780b2), which we could also implement
![image](https://github.com/Yooooomi/your_spotify/assets/58067119/5e333fc7-ab23-443a-8cb2-fd6ed5a7a10e)

And a interesting [article](https://archives.ismir.net/ismir2021/latebreaking/000003.pdf) over the additional audio-analysis

**Comment by Yooooomi** [comment]
Yeah I see the type of graph you can do. I honestly feel in your example that it's wow but in the actual usage it won't be very interesting you know. I'll check your current work regarding the addition of audio stats in the database.

**Comment by qdlmcfresh** [comment]
I think stats for genres would also be a great addition (like [Stats for Spotify](https://www.statsforspotify.com/) has it).

**Comment by Matoxius** [comment]
I second the genres Stats, I hear a lot of different genres and really would appreciate a graph that shows the most listened genres in a specific timeframe (Day/Month/Year etc.).

**Comment by qdlmcfresh** [comment]
> I think stats for genres would also be a great addition (like [Stats for Spotify](https://www.statsforspotify.com/) has it).

@Yooooomi 
I'm currently looking into what would be needed to extend `your_spotify` with a "Top Genres" tab. However, I've encountered a few challenges:

1. The only Spotify API endpoint that includes genres is the album endpoint. Unfortunately, this limits our ability to directly access genre data for other entities like tracks or artists.

2. Upon inspecting the MongoDB collection, I noticed that while the `genre` field does exist for albums, every genre array is empty in my case. 


![image](https://github.com/user-attachments/assets/47e4a192-e259-452a-be4a-b25a9166ee3c)
